spec/std/uri_spec.cr

```diff
@@ -111,35 +212,21 @@ describe "URI" do
 
   describe "to_s" do
     it { URI.new("http", "www.example.com").to_s.should eq("http://www.example.com") }
-    it { URI.new("http", "www.example.com", 80).to_s.should eq("http://www.example.com") }
-    it do
-      u = URI.new("http", "www.example.com")
-      u.user = "alice"
-      u.to_s.should eq("http://alice@www.example.com")
-      u.password = "s3cr3t"
-      u.to_s.should eq("http://alice:s3cr3t@www.example.com")
-    end
-    it do
-      u = URI.new("http", "www.example.com")
-      u.user = ":D"
-      u.to_s.should eq("http://%3AD@www.example.com")
-      u.password = "@_@"
-      u.to_s.should eq("http://%3AD:%40_%40@www.example.com")
-    end
+    it { URI.new("http", "www.example.com", 80).to_s.should eq("http://www.example.com:80") }
+    it { URI.new("http", "www.example.com", user: "alice").to_s.should eq("http://alice@www.example.com") }
+    it { URI.new("http", "www.example.com", user: "alice", password: "s3cr3t").to_s.should eq("http://alice:s3cr3t@www.example.com") }
+    it { URI.new("http", "www.example.com", user: ":D").to_s.should eq("http://%3AD@www.example.com") }
+    it { URI.new("http", "www.example.com", user: ":D", password: "@_@").to_s.should eq("http://%3AD:%40_%40@www.example.com") }
     it { URI.new("http", "www.example.com", user: "@al:ce", password: "s/cr3t").to_s.should eq("http://%40al%3Ace:s%2Fcr3t@www.example.com") }
     it { URI.new("http", "www.example.com", fragment: "top").to_s.should eq("http://www.example.com#top") }
-    it { URI.new("http", "www.example.com", 80, "/hello").to_s.should eq("http://www.example.com/hello") }
-    it { URI.new("http", "www.example.com", 80, "/hello", "a=1").to_s.should eq("http://www.example.com/hello?a=1") }
-    it { URI.new("mailto", opaque: "foo@example.com").to_s.should eq("mailto:foo@example.com") }
-
-    it "removes default port" do
```

Comment:
Why not normalize ports from default?

Reply:
There are no generic default ports in the URI specification. It depends solely on configuration which port is considered "default".

When no port is specified, the resolver is entrusted to choose whichever the port it seems fit. But when a port is explicitly specified, the resolver is asked to use that specific port, regardless of it's configuration for default ports.

Thus removing a port (even if it is considered default) mutates the URI. It looses information.
A well-formed URI should not be automatically stripped of anything unless explicitly asked (for example using `#normalize´).